src/scripting/lua_game.cc

```diff
@@ -690,8 +693,11 @@ int LuaPlayer::hide_fields(lua_State* L) {
 	Widelands::Player& p = get(L, game);
 
 	luaL_checktype(L, 2, LUA_TTABLE);
-	const Widelands::HideOrRevealFieldMode mode = (!lua_isnone(L, 3) && luaL_checkboolean(L, 3)) ?
+	const std::string state = luaL_checkstring(L, 3);
+	const Widelands::HideOrRevealFieldMode mode = (!lua_isnone(L, 3) && state == "permanent") ?
 	                                                 Widelands::HideOrRevealFieldMode::kHide :
+	                                                 (!lua_isnone(L, 3) && state == "explorable") ?
+	                                                 Widelands::HideOrRevealFieldMode::kUnexplore :
 	                                                 Widelands::HideOrRevealFieldMode::kUnreveal;
```

Comment:
I guess here's the problem. You assume there's a string in position 3 before checking whether position 3 is nil.
```suggestion
	const std::string state = lua_isnone(L, 3) ? "SOME_DEFAULT_VALUE" : luaL_checkstring(L, 3);
	const Widelands::HideOrRevealFieldMode mode = (state == "permanent") ?
	                                                 Widelands::HideOrRevealFieldMode::kHide :
	                                                 (state == "explorable") ?
	                                                 Widelands::HideOrRevealFieldMode::kUnexplore :
	                                                 Widelands::HideOrRevealFieldMode::kUnreveal;
```

Reply:
*Head Desk* you are right thanks for the hint